compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -2487,6 +2487,12 @@ impl<'tcx> TyCtxt<'tcx> {
                 .hygienic_eq(def_name.span.ctxt(), self.expn_that_defined(def_parent_def_id))
     }
 
+    pub fn adjust_span(self, span: Span) -> Span {
```

Comment:
When should one use `adjust_span`? (needs some docs on this function)

All these call sites make me think that this is a bit fragile, if we forget it somewhere, a diagnostic will be subpar and it will be hard to find out where the problematic span is from. Can we do this in a more principled way? Like somewhere during ast lowering or even macro expansion?